lib/request.test.js

```diff
@@ -0,0 +1,160 @@
+const getTest = {
+  url: '/market-data/getUsdtPrice/USDT',
+  response: {
+    price: 1,
+    token: 'USDT'
+  }
+}
+
+// Post Test Url / Response
+const postTest = {
+  url: '/v1/trading/r/getConf',
+  response: {
+    DVF: {
+      defaultFeeRate: 0.002,
+      defaultFeeRateSwap: 0.003
+    }
+  }
+}
+
+describe('Request Tests', () => {
+  let requestNock
+
+  beforeAll(() => {
+    nock.cleanAll()
+  })
+
+  beforeEach(() => {
+    requestNock = nock(apiDomain).defaultReplyHeaders({
+      'access-control-allow-origin': '*'
+    })
+  })
+
+  afterEach(() => {
+    nock.cleanAll()
+  })
+
+  afterAll(() => {
+    nock.cleanAll()
+  })
+
+  describe('Expected Success', () => {
+    it('returns default data only for get', async () => {
+      requestNock.get(getTest.url).reply(200, getTest.response)
+
+      const response = await request.get(`${apiDomain}${getTest.url}`)
+
+      expect(response).toEqual(getTest.response)
+    })
+
+    it('returns default data only for post', async () => {
+      requestNock.post(postTest.url).reply(200, postTest.response)
+
+      const response = await request.post(`${apiDomain}${postTest.url}`)
+
+      expect(response).toEqual(postTest.response)
+    })
+
+    it('returns extended data using fullResponse for get', async () => {
+      requestNock.get(getTest.url).reply(200, getTest.response)
+
+      const response = await request.get(`${apiDomain}${getTest.url}`, {
+        fullResponse: true
+      })
+
+      expect(response).not.toEqual(getTest.response)
+      expect(response.data).toEqual(getTest.response)
+      expect(response.statusCode).toBe(200)
+      expect(response.statusText).toBe('OK')
+      expect(response.headers).toBeDefined()
+    })
+
+    it('returns extended data using fullResponse for post', async () => {
+      requestNock.post(postTest.url).reply(200, postTest.response)
+
+      const response = await request.post(`${apiDomain}${postTest.url}`, {
+        fullResponse: true
+      })
+
+      expect(response).not.toEqual(postTest.response)
+      expect(response.data).toEqual(postTest.response)
+      expect(response.statusCode).toBe(200)
+      expect(response.statusText).toBe('OK')
+      expect(response.headers).toBeDefined()
+    })
+
+    it('returns plain text data using responseType text', async () => {
+      requestNock.get(getTest.url).reply(200, JSON.stringify(getTest.response))
+
+      // Issue with plain text, sometimes works, sometimes not
+      const response = await request.get(`${apiDomain}${getTest.url}`, {
+        responseType: 'text'
+      })
+
+      expect(response).not.toEqual(getTest.response)
+    })
+  })
+
+  describe('Expected Failure', () => {
+    it('fails with InvalidOptionError using data option for get', async () => {
+      expect(async () => {
+        await request.get(`${apiDomain}${getTest.url}`, {
+          data: {
+            test: 'test'
+          }
+        })
+      }).rejects.toThrow(InvalidOptionError)
```

Comment:
`expect().rejects` returns a promise so it needs to be awaited or returned 

and it can also be passed a promise (instead of function returning a promise), so the whole thing could be slightly simplified 

```suggestion
    it('fails with InvalidOptionError using data option for get', () => {
      return expect(request.get(`${apiDomain}${getTest.url}`, {
        data: {
          test: 'test'
        }
      })).rejects.toThrow(InvalidOptionError)
    })
```